XamlAnimatedGif/TimingManager.cs

```diff
@@ -102,6 +102,11 @@ private set
         public void Pause()
         {
             IsPaused = true;
+            if (_pauseCompletionSource != null) 
+            {
+                var tcs = _pauseCompletionSource;
+                tcs?.TrySetResult(0); 
+            }
```

Comment:
Is this actually necessary? It seems to me that calling `Pause()` when `IsPaused` is already true should just do nothing. I think `if (IsPaused) return;` would probably be enough to solve the problem.

Reply:
(in fact, there might be a race condition if two threads try to pause at the same time, but it doesn't seem very likely, so I'm not sure it's worth the trouble of guarding against it...)

Reply:
You wrote this, so I'm sure you know if there are any _other_ situations where not setting a result would be a problem.
I'm going to take your word for it.   

The setting a result prior to the new TaskCompletionSource is the safer way to do it when I don't know the codebase.
I will close this PR and open a new one with your recommendation.

Reply:
> I will close this PR and open a new one with your recommendation.

You didn't need to open a new PR, you could have just pushed a new commit to this one ;)